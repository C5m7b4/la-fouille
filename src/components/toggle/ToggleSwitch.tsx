import React from 'react';
import './ToggleSwitch.css';

export interface IToggleProps {
  active: boolean;
  handleChange: () => void;
  mainDivClassName?: string;
  mainDivStyle?: object;
  inputClassName?: string;
  inputStyle?: object;
  labelClassName?: string;
  labelStyle?: object;
  labelSpan1ClassName?: string;
  labelSpan2ClassName?: string;
}

const ToggleSwitch = (props: IToggleProps) => {
  const {
    handleChange,
    active,
    mainDivClassName = 'toggle-switch',
    mainDivStyle,
    inputClassName = 'toggle-switch-checkbox',
    inputStyle,
    labelClassName = 'toggle-switch-label',
    labelSpan1ClassName = 'toggle-switch-inner',
    labelSpan2ClassName = 'toggle-switch-switch',
  } = props;

  return (
    <div
      className={mainDivClassName}
      style={mainDivStyle}
      onClick={handleChange}
    >
      <input
        type="checkbox"
        className={inputClassName}
        style={inputStyle}
        onChange={handleChange}
        checked={active}
      />
      <label className={labelClassName}>
        <span className={labelSpan1ClassName} />
        <span className={labelSpan2ClassName} />
      </label>
    </div>
  );
};

export default ToggleSwitch;
