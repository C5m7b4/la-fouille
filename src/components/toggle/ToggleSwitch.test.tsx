import React from 'react';
import ToggleSwitch from './ToggleSwitch';
import { render } from '@testing-library/react';

describe('Toggle Switch Component', () => {
  it('should render without crashing', () => {
    const { asFragment } = render(
      <ToggleSwitch active={true} handleChange={() => {}} />
    );

    expect(asFragment).toMatchSnapshot();
  });
});
