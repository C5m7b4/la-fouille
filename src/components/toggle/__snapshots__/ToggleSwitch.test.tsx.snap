// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toggle Switch Component should render without crashing 1`] = `[Function]`;
